crates/rust-analyzer/src/config.rs

```diff
@@ -196,110 +175,113 @@ config_data! {
+        /// Whether to render trailing colons for parameter hints, and trailing colons for parameter hints.
+        inlayHints_renderColons: bool                      = "true",
+        /// Whether to show inlay type hints for variables.
+        inlayHints_typeHints_enable: bool                          = "true",
         /// Whether to hide inlay hints for constructors.
-        inlayHints_hideNamedConstructorHints: bool          = "false",
+        inlayHints_typeHints_hideNamedConstructor: bool          = "false",
 
+        /// Join lines merges consecutive declaration and initialization of an assignment.
+        joinLines_joinAssignments: bool = "true",
         /// Join lines inserts else between consecutive ifs.
         joinLines_joinElseIf: bool = "true",
         /// Join lines removes trailing commas.
         joinLines_removeTrailingComma: bool = "true",
         /// Join lines unwraps trivial blocks.
         joinLines_unwrapTrivialBlock: bool = "true",
-        /// Join lines merges consecutive declaration and initialization of an assignment.
-        joinLines_joinAssignments: bool = "true",
 
         /// Whether to show `Debug` lens. Only applies when
         /// `#rust-analyzer.lens.enable#` is set.
-        lens_debug: bool            = "true",
+        lens_debug_enable: bool            = "true",
         /// Whether to show CodeLens in Rust files.
         lens_enable: bool           = "true",
+        /// Internal config: use custom client-side commands even when the
+        /// client doesn't set the corresponding capability.
+        lens_forceCustomCommands: bool = "true",
         /// Whether to show `Implementations` lens. Only applies when
         /// `#rust-analyzer.lens.enable#` is set.
-        lens_implementations: bool  = "true",
-        /// Whether to show `Run` lens. Only applies when
-        /// `#rust-analyzer.lens.enable#` is set.
-        lens_run: bool              = "true",
-        /// Whether to show `Method References` lens. Only applies when
-        /// `#rust-analyzer.lens.enable#` is set.
-        lens_methodReferences: bool = "false",
-        /// Whether to show `References` lens for Struct, Enum, Union and Trait.
+        lens_implementations_enable: bool  = "true",
+        /// Whether to show `References` lens for Struct, Enum, and Union.
         /// Only applies when `#rust-analyzer.lens.enable#` is set.
-        lens_references: bool = "false",
+        lens_references_adt_enable: bool = "false",
         /// Whether to show `References` lens for Enum Variants.
         /// Only applies when `#rust-analyzer.lens.enable#` is set.
-        lens_enumVariantReferences: bool = "false",
-        /// Internal config: use custom client-side commands even when the
-        /// client doesn't set the corresponding capability.
-        lens_forceCustomCommands: bool = "true",
+        lens_references_adt_variant_enable: bool = "false",
```

Comment:
Unsure if this should really go under `adt`

Reply:
That sounds like it enables it for `struct` and `union` too, so maybe move it out?